openssl-dynamic/src/main/native-package/configure.ac

```diff
@@ -29,12 +29,14 @@ AC_CANONICAL_HOST
 AC_CANONICAL_SYSTEM
 
 ${CFLAGS="-O3 -Werror -fno-omit-frame-pointer -Wunused-variable"}
-${CXXFLAGS="-O3 -Werror -fno-omit-frame-pointer -Wunused-variable"}
+# Ensure we support c++14 for atomics etc
+${CXXFLAGS="-O3 -Werror -fno-omit-frame-pointer -Wunused-variable -std=c++14"}
```

Comment:
I'm not sure if the situation has changed, but I had an impression that libstdc++ has worse backward ABI compatibility historically. Maybe I'm completely wrong, though.